### PR TITLE
[Core][Minor]Remove the hard check when disconnect gcs client

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
There's a hard flag check when we close the gcs client but sometimes we would disconnect the gcs client for more than one time because the gcs client may be shared between different objects(now in accessor and core worker).
In case of unwanted FATAL error and exit, I'd like to remove the hard check and rephrase it to a warning log.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jovany-wang — APPROVED** [review]

**Comment by WangTaoTheTonic** [comment]
```
>       assert proc.port == 45000

E       assert 45001 == 45000

E         +45001

E         -45000
```
There's one only test failure and seems like nothing related with this change but port conflicts.

**Comment by jovany-wang** [comment]
I believe the CI failure is unrelated to this change.

@ijrsvt Can you take a look to confirm that?

**Comment by ijrsvt** [comment]
@jovany-wang Yep! That looks unrelated to this change!

**Review by rkooo567 — APPROVED** [review]
